engine/baml-lib/parser-database/src/attributes/constraint.rs

```diff
@@ -1,38 +1,56 @@
 use baml_types::{Constraint, ConstraintLevel};
-use internal_baml_schema_ast::ast::Expression;
+use internal_baml_diagnostics::{DatamodelError, Span};
+use internal_baml_schema_ast::ast::{Attribute, Expression};
 
 use crate::{context::Context, types::Attributes};
 
 pub(super) fn visit_constraint_attributes(
     attribute_name: String,
+    span: Span,
     attributes: &mut Attributes,
     ctx: &mut Context<'_>,
 ) {
-    let expression_arg = ctx.visit_default_arg_with_idx("expression").map_err(|err| {
-        ctx.push_error(err);
-    });
-    let label = ctx.visit_default_arg_with_idx("name");
-    let label = match label {
-        Ok((_, Expression::StringValue(descr, _))) => Some(descr.clone()),
-        _ => None,
+    let arguments: Vec<&Expression> = ctx
+        .get_all_args()
+        .into_iter()
+        .map(|(_arg_id, arg)| arg)
+        .collect();
+
+    let level = match attribute_name.as_str() {
+        "assert" => ConstraintLevel::Assert,
+        "check" => ConstraintLevel::Check,
+        _ => panic!("Internal error - the parser should have ruled out other attribute names."),
```

Comment:
do we need to panic here? what if there was a ctx.push_error and we just returned?

Reply:
Good point. Replaced it with a diagnostics error and short-circuit return.